light-client/src/fork_detector.rs

```diff
@@ -109,8 +112,16 @@ impl ForkDetector for ProdForkDetector {
                 .verify_to_target(light_block.height(), &mut state);
 
             match result {
-                Ok(_) => forks.push(Fork::Forked(witness_block)),
-                Err(e) if e.kind().has_expired() => forks.push(Fork::Forked(witness_block)),
+                Ok(_) => forks.push(Fork::Forked {
+                    primary: light_block.clone(),
+                    witness: witness_block,
+                }),
+                Err(e) if e.kind().has_expired() => {
```

Comment:
```suggestion
                Err(e) if !e.kind().has_expired() => {
```

?

Reply:
Could you please expand on that?

The spec currently describe the case where header expired as a potential fork: https://github.com/informalsystems/tendermint-rs/pull/287/files#diff-57dd6c1989dd85c55b47487b7e1f92d8R409-R414

Reply:
Right. We report it even though we won't be able to punish. All good